### PR TITLE
Issue #22489: Remove "Fennec to Fenix" migration code

### LINKED ISSUE #22489: Investigate removing Fennec -> Fenix migration code
We are contemplating removing the "Fennec to Fenix" migration code from Fenix. We shipped the first Fenix release with migration code over a year ago (~ Summer 2020). Removing the code will reduce the maintenance burden, improve performance (#15581) and reduce the APK size.
We still need to go through the process of looking at data, getting approval etc. - but that doesn't stop us from investigating the engineering work already.

### PR DESCRIPTION
**Pull request by gabrielluong**
Fixes #22489. This is a rebase of https://github.com/mozilla-mobile/fenix/pull/22507 mainly to address any conflicts and bump strings removedIn version.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by gabrielluong** [comment]
@grigoryk @csadilek @jonalmeida Let's figure out what we need to do to finally remove this, but feel free to take over.

**Comment by csadilek** [comment]
@gabrielluong all that's left is review and land. we have sign-off to land when ready.

**Comment by gabrielluong** [comment]
> @gabrielluong all that's left is review and land. we have sign-off to land when ready.

Based on https://github.com/mozilla-mobile/fenix/pull/22507#issue-1058539360, I imagine we also want to test thoroughly, but I am not quite sure what I should be looking for.

**Review by grigoryk — COMMENTED** [review]
Just looking through the changes, it all looks fine at first glance. I think we mostly just need to make ensure that nothing odd happens w/ things such as shortcuts, PWAs, search widget, etc. 

What kind of local testing did you do for this?

To get started, I'm going to try the following cases locally:
- verify that home screen shortcuts, installed PWA, search widget continue to work (when installed from fenix, and when installed from fennec and then migrated)
- install fennec release variant, upgrade to fenix (w/ migration code in place), upgrade to fenix (w/o migration code)
- do the same for other channels (beta, nightly, i think that's it?)

**Review by grigoryk — APPROVED** [review]
👍 🚢

File: app/build.gradle [hunk 1]
```diff
@@ -177,14 +177,6 @@ android {
         androidTest {
             resources.srcDirs += ['src/androidTest/resources']
         }
-        beta {
-            java.srcDirs = ['src/migration/java']
-            manifest.srcFile "src/migration/AndroidManifest.xml"
```
[the remark this answers is not in the document]
Comment (gabrielluong) [reply app/build.gradle:182]: Yes

**Pull request merged**